### PR TITLE
feat(check_recipe_name): Refactor and move sub

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR fixes:

- See above 

### How to test:

- Automatic and continuous test pass

### Expected outcome:
- Installation, unit and integration tests pass

### Review:
- [ ] Code review
- [ ] New code is executed and covered by tests
- [ ] Tests pass

### CODE REVIEW

File: lib/MIP/Parameter.pm [hunk 1]
```diff
@@ -122,6 +123,60 @@ sub check_parameter_hash {
...
+  RECIPE:
+    foreach my $recipe ( @{$recipe_names_ref} ) {
+
+        next RECIPE if ( not exists $parameter_href->{$recipe}{program_executables} );
+
+      BINARY:
+        foreach my $binary ( @{ $parameter_href->{$recipe}{program_executables} } ) {
+
+            $binary_name{$binary} = undef;
+        }
```
Comment (jemten) [inline lib/MIP/Parameter.pm]: There's more than one way to do it, but couldn't this be written as:
```Perl
@binary_name{ @{ $parameter_href->{$recipe}{program_executables} } = undef;
```
to save some much valued lines of code ;)

File: lib/MIP/Parameter.pm [hunk 2]
```diff
@@ -122,6 +123,60 @@ sub check_parameter_hash {
...
+
+        next RECIPE if ( not exists $binary_name{$recipe} );
```
Comment (jemten) [inline lib/MIP/Parameter.pm]: Isn't it better to check the binary name hash for matching recipe name after it has been fully populated? We could potentially have a recipe named after an executable that hasn't been added to the hash yet.

File: lib/MIP/Parameter.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Parameter.pm]: You are right, although it looks a bit heavy for a oneliner. However, but combining your two excellent comments - created another solution 😄. I do not think I managed to save lines, but the functionality is extended and easy to follow (and probably faster...)

File: lib/MIP/Parameter.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Parameter.pm]: I pondered the same thing when I refactored this one. Let´s do it.

**Comment by henrikstranneheim** [comment]
What do you think of this proposed solution?

**Comment by jemten** [comment]
Looks good! The get_cache sub certainly makes it more readable than my somewhat heavy handed oneliner 😄

**Pull request merged**